shell/browser/api/electron_api_web_contents.cc

```diff
@@ -3236,6 +3236,32 @@ v8::Local<v8::Promise> WebContents::TakeHeapSnapshot(
   return handle;
 }
 
+void WebContents::GrantDevicePermission(
+    const url::Origin& origin,
+    const base::Value* device,
+    content::PermissionType permissionType) {
+  granted_devices_[permissionType][origin].push_back(
```

Comment:
We probably want to clear `granted_devices_` when navigation occurs. Perhaps there's a different place we could store this set that 'naturally' gets reset already when navigation happens?